--- ../gcc-7.3.0/contrib/download_prerequisites	2019-04-20 11:34:51.893245009 +0200
+++ contrib/download_prerequisites	2019-04-20 11:35:46.318173531 +0200
@@ -32,7 +32,7 @@
 mpc='mpc-1.0.3.tar.gz'
 isl='isl-0.16.1.tar.bz2'
 
-base_url='ftp://gcc.gnu.org/pub/gcc/infrastructure/'
+base_url='http://gcc.gnu.org/pub/gcc/infrastructure/'
 
 echo_archives() {
     echo "${gmp}"
@@ -219,7 +219,7 @@
 do
     if [ ${force} -gt 0 ]; then rm -f "${directory}/${ar}"; fi
     [ -e "${directory}/${ar}" ]                                               \
-        || ${fetch} --no-verbose -O "${directory}/${ar}" "${base_url}${ar}"       \
+        || ${fetch} -O "${directory}/${ar}" "${base_url}${ar}"       \
         || die "Cannot download ${ar} from ${base_url}"
 done
 unset ar
